File: codeforces/acmsguru/verySimpleProblem/problem.py
import math
x = int(input())
def getAns():
    lo = 1
    hi = x
    while (lo + 1 < hi):
        mid = (lo+hi)//2
        if (mid*mid) < x:
            lo = mid
        elif (mid*mid > x):
            hi = mid
        elif (mid*mid == x):
            return mid
    return lo
print(getAns())

File: codeforces/acmsguru/987654321Problem/problem.cpp
#include <iostream>
#include <math.h>
using namespace std;

int main()
{
    for (long long i = 100000000; i < 1000000000; i++) {
        if (i*i % 1000000000 == 987654321) {
            cout << i << ' ' << i*i << '\n';
        }
    }
}
